R/SEIRDAge.R

```diff
@@ -0,0 +1,328 @@
+    }
+    
+    #fetch number of age catagories
+    age <- object@n_age_categories
+
+    # set initial state vector
+    state <- c(S = initial_conditions(object)$S0,
+               E = initial_conditions(object)$E0,
+               I = initial_conditions(object)$I0,
+               R = initial_conditions(object)$R0,
+               D = initial_conditions(object)$D0)
+
+    # set parameters vector
+    parameters <- c(b = transmission_parameters(object)$b,
+                    k = transmission_parameters(object)$k,
+                    g = transmission_parameters(object)$g,
+                    mu = transmission_parameters(object)$mu)
+    
+    # fetch contact matrix of the instance
+    C = object@contact_matrix
+
+    # function for RHS of ode system
+    right_hand_side <- function(t, state, parameters) {
+      with(
+        as.list(c(state, parameters)),
+        {
+          S <- state[1:age]
+          E <- state[(age + 1):(2 * age)]
+          I <- state[(2 * age + 1):(3 * age)]
+          R <- state[(3 * age + 1):(4 * age)]
+          D <- state[(4 * age + 1):(5 * age)]
+          
+          
+          # rate of change
+          dS <- -b * S * C %*% I
+          dE <- b * S * C %*% I - k * E
+          dI <- k * E - g * I
+          dR <- g * I
+          dD <- mu * I
+          # return the rate of change
+          list(c(dS, dE, dI, dR, dD))
+        })
+    }
+
+    # call ode solver
+    out <- ode(
+      y = state, times = times, func = right_hand_side,
+      parms = parameters, method = solve_method)
+
+    #output as a dataframe
+    output <- as.data.frame.array(out)
+
+    # melt dataframe to wide format
+    out_temp = melt(output, 'time')
+
+    # add compartment and age range columns
+    out_temp$compartment = c(replicate(length(times)*age, "S"),
+                             replicate(length(times)*age, "E"),
+                             replicate(length(times)*age, "I"),
+                             replicate(length(times)*age, "R"),
+                             replicate(length(times)*age, "D"))
+
+    out_temp$age_range = unlist(rep(object@age_ranges, each=length(times)))
+
+    #drop the old variable column
+    out_temp = out_temp %>% 
+      dplyr::select(-.data$variable) %>% 
+      dplyr::mutate(compartment=as.factor(compartment)) %>% 
+      dplyr::mutate(compartment=forcats::fct_relevel(compartment, "S", "E", "I", "R","D")) %>% 
+      dplyr::mutate(age_range=as.factor(age_range)) %>% 
+      dplyr::mutate(age_range=forcats::fct_relevel(age_range, object@age_ranges))
+
+    # compute incidence number
+    total_inf <- output[, (2*age+2):(3*age+1)] + output[, (3*age+2):(4*age+1)]
+    n_inc <- rbind(rep(0, age),
+                   total_inf[2:nrow(total_inf),]-total_inf[1:nrow(total_inf)-1,]
+                   )
+
+    # melt the incidence dataframe to long format
+    incidence_temp = melt(n_inc, id.vars=NULL)
+    print(incidence_temp)
```

Comment:
Is this print intended?

Reply:
I tried to fins sth for this one, but could not find anything useful for the docstring part. If it were code, a ``paste0`` would have worked